src/wp-includes/css/admin-bar.css

```diff
@@ -719,6 +724,50 @@ html:lang(he-il) .rtl #wpadminbar * {
 	box-shadow: 0 0 2px 2px rgba(0, 0, 0, 0.6);
 }
 
+@media only screen and (min-width: 1061px) {
+	/* Reset the spacing */
+	#wpadminbar .quicklinks > ul > li,
+	#wpadminbar .quicklinks .ab-item {
+		max-width: initial;
+		text-overflow: clip;
+	}
+}
+@media screen and (min-width: 782px) {
```

Comment:
This is really the heart of this update. Utilize flexbox on the admin bar where it allows plugins to add items to it so that it never overflows onto a new line (instead, scaling proportionally and using an ellipses-based text overflow when needed & looking effectively the same as before when not needed.)